portal/static/js/i18next-config.js

```diff
@@ -21,110 +21,114 @@ var __i18next = window.__i18next = (function() {
+                } catch (e) {
+                    return key;
+                }
+                return key;
+            },
+            backend: {
+                // load from static file
+                language: options.lng,
+                loadPath: source,
+                ajax: function(url, options, callback, data, cache) {
                     /*
-                     * allow lookup for translated text for missing key
+                     * default code from i18nextBackend.js, but modify it to allow sync loading of resources and add session storage
                      */
-                    var sessionData = sessionStorage.getItem("i18nextData_" + this.lng);
-                    if (sessionData) {
-                        var data;
-                        try {
-                            data = JSON.parse(sessionData);
-                            if (data && data.hasOwnProperty(key)) {
-                                return data[key];
-                            }
-                        } catch (e) {
-                            return key;
-                        }
+                    callback = callback || function() {};
+                    if (options.language === "en-US") {
+                        callback();
+                        return false;
                     }
-                    return key;
-                },
-                backend: {
-                    // load from static file
-                    language: options.lng,
-                    loadPath: source,
-                    ajax: function(url, options, callback, data, cache) {
-                        /*
-                         * default code from i18nextBackend.js, but modify it to allow sync loading of resources and add session storage
-                         */
-                        var sessionItemKey = "i18nextData_" + options.language;
-                        if (data && (typeof data === 'undefined' ? 'undefined' : _typeof(data)) === 'object') { /*global _typeof */
-                            if (!cache) { data["_t"] = new Date();}
-                            // URL encoded form data must be in querystring format
-                            data = addQueryString("", data).slice(1); /* global addQueryString */
+                    var sessionItemKey = "i18nextData_" + options.language;
+                    if (data && (typeof data === "undefined" ? "undefined" : _typeof(data)) === "object") { /*global _typeof */
+                        if (!cache) { data["_t"] = new Date();}
+                        // URL encoded form data must be in querystring format
+                        data = addQueryString("", data).slice(1); /* global addQueryString */
+                    }
+                    if (options.queryStringParams) {
+                        url = addQueryString(url, options.queryStringParams); /* global addQueryString */
+                    }
+                    try {
```

Comment:
Are you able to move some of this code outside of the `try` block (unless all of it could throw an exception)?